tests/src/python/test_qgsvectorlayer.py

```diff
@@ -2091,6 +2099,221 @@ def testGetFeatureLimitWithEdits(self):
         self.assertEqual(len(list(layer.getFeatures(req))), 2)
         layer.rollBack()
 
+    def testCloneMapLayerAttributes(self):
+        # init crs
+        srs = QgsCoordinateReferenceSystem(3111, QgsCoordinateReferenceSystem.EpsgCrsId)
+
+        # init map layer styles
+        tmplayer = createLayerWithTwoPoints()
+        sym1 = QgsLineSymbol()
+        sym1.setColor(Qt.magenta)
+        tmplayer.setRenderer(QgsSingleSymbolRenderer(sym1))
+
+        style0 = QgsMapLayerStyle()
+        style0.readFromLayer(tmplayer)
+        style1 = QgsMapLayerStyle()
+        style1.readFromLayer(tmplayer)
+
+        # init dependencies layers
+        ldep = createLayerWithTwoPoints()
+        dep = QgsMapLayerDependency(ldep.id())
+
+        # init layer
+        layer = createLayerWithTwoPoints()
+        layer.setBlendMode(QPainter.CompositionMode_Screen)
+        layer.styleManager().addStyle('style0', style0)
+        layer.styleManager().addStyle('style1', style1)
+        layer.setName('MyName')
+        layer.setShortName('MyShortName')
+        layer.setMinimumScale(0.5)
+        layer.setMaximumScale(1.5)
+        layer.setScaleBasedVisibility(True)
+        layer.setTitle('MyTitle')
+        layer.setAbstract('MyAbstract')
+        layer.setKeywordList('MyKeywordList')
+        layer.setDataUrl('MyDataUrl')
+        layer.setDataUrlFormat('MyDataUrlFormat')
+        layer.setAttribution('MyAttribution')
+        layer.setAttributionUrl('MyAttributionUrl')
+        layer.setMetadataUrl('MyMetadataUrl')
+        layer.setMetadataUrlType('MyMetadataUrlType')
+        layer.setMetadataUrlFormat('MyMetadataUrlFormat')
+        layer.setLegendUrl('MyLegendUrl')
+        layer.setLegendUrlFormat('MyLegendUrlFormat')
+        layer.setDependencies([dep])
+        layer.setCrs(srs)
+
+        layer.setCustomProperty('MyKey0', 'MyValue0')
+        layer.setCustomProperty('MyKey1', 'MyValue1')
+
+        # clone
+        clone = layer.clone()
+
+        # test QgsMapLayer attributes
```

Comment:
Maybe it would be maybe easier to call writeLayerXml() on both original and cloned layer (with patched layer ID to match original layer) to see if they are the same...